core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -446,6 +446,16 @@ public QuerySearchResult executeQueryPhase(QuerySearchRequest request) {
         }
     }
 
+    private boolean fetchPhaseShouldFreeContext(SearchContext context) {
+        if (context.scrollContext() == null) {
+            // simple search, no scroll
+            return true;
+        } else {
+            // scroll request, but the scroll was not extended
```

Comment:
I think we can do it, but this would have a 3.0 change as otherwise the call to clear the scroll would suddenly return a 4xx code instead of a 2xx code.